api/pkg/apis/v1alpha1/utils/apiclient.go

```diff
@@ -98,6 +98,28 @@ type (
 	}
 )
 
+// We shouldn't use specific error types
+// APIError represents an error that includes a SummarySpec in its message field.
+type APIError struct {
+	Code    v1alpha2.State `json:"code"`
+	Message string         `json:"message"`
+}
+
+func (e APIError) Error() string {
+	return fmt.Sprintf(
+		"failed to invoke Symphony API: [%v] - %s",
+		e.Code,
+		e.Message,
+	)
+}
+
+func NewAPIError(state v1alpha2.State, msg string) APIError {
+	return APIError{
+		Code:    state,
+		Message: msg,
+	}
+}
+
```

Comment:
Could you add a helper function to convert API errors to Coa errors. It is useful when the service gets an API error from http server but need to return Coa error.

Reply:
updated.